crates/astria-bridge-withdrawer/src/bridge_withdrawer/startup.rs

```diff
@@ -0,0 +1,614 @@
+            .wrap_err("failed to convert last transaction hash to tendermint hash")?;
+
+        // get the corresponding transaction
+        let last_transaction = get_tx(
+            self.sequencer_cometbft_client.clone(),
+            self.state.clone(),
+            tx_hash,
+        )
+        .await
+        .wrap_err("failed to fetch last transaction by the bridge account")?;
+
+        // check that the transaction actually executed
+        ensure!(
+            last_transaction.tx_result.code == tendermint::abci::Code::Ok,
+            "last transaction by the bridge account failed to execute. this should not happen in \
+             the sequencer logic."
+        );
+
+        let proto_tx =
+            astria_core::generated::protocol::transaction::v1alpha1::SignedTransaction::decode(
+                &*last_transaction.tx,
+            )
+            .wrap_err("failed to convert transaction data from CometBFT to proto")?;
+
+        let tx = SignedTransaction::try_from_raw(proto_tx)
+            .wrap_err("failed to convert transaction data from proto to SignedTransaction")?;
+
+        info!(
+            last_bridge_account_tx.hash = %telemetry::display::hex(&tx_hash),
+            last_bridge_account_tx.height = i64::from(last_transaction.height),
+            "fetched last transaction by the bridge account"
+        );
+
+        Ok(Some(tx))
+    }
+
+    /// Gets the data necessary for syncing to the latest on-chain state from the sequencer.
+    /// Since we batch all events from a given rollup block into a single sequencer
+    /// transaction, we get the last tx finalized by the bridge account on the sequencer
+    /// and extract the rollup height from it.
+    ///
+    /// The rollup height is extracted from the block height value in the memo of one of the
+    /// actions in the batch.
+    ///
+    /// # Returns
+    /// The next batch rollup height to process.
+    ///
+    /// # Errors
+    ///
+    /// 1. Failing to get and deserialize a valid last transaction by the bridge account from the
+    ///    sequencer.
+    /// 2. The last transaction by the bridge account failed to execute (this should not happen in
+    ///    the sequencer logic)
+    /// 3. The last transaction by the bridge account did not contain a withdrawal action
+    /// 4. The memo of the last transaction by the bridge account could not be parsed
+    async fn get_starting_rollup_height(&mut self) -> eyre::Result<u64> {
+        let signed_transaction = self
+            .get_last_transaction()
+            .await
+            .wrap_err("failed to get the bridge account's last sequencer transaction")?;
+        let starting_rollup_height = if let Some(signed_transaction) = signed_transaction {
+            rollup_height_from_signed_transaction(&signed_transaction).wrap_err(
+                "failed to extract rollup height from last transaction by the bridge account",
+            )?
```

Comment:
```suggestion
            rollup_height_from_signed_transaction(&signed_transaction).wrap_err(
                "failed to extract rollup height from last transaction by the bridge account",
            )? + 1
```

this returns the last submitted rollup height, but we want to return the next rollup height to process, which is last submitted + 1.

Reply:
good catch, my bad